igneous/tasks/skeletonization/skeletonization.py

```diff
@@ -81,24 +83,26 @@ def TEASAR(
   # compute multiple paths by simply hopping pointers using path_from_parents
   parents = igneous.dijkstra.parental_field(PDRF, root)
 
-  if dbf_max > soma_detection_threshold:
+  invalid_vertices = {}
+  if soma_mode:
       invalidated, labels = igneous.skeletontricks.roll_invalidation_ball(
-      labels, DBF, np.array([root], dtype=np.uint32),
-      scale=soma_invalidation_scale,
-      const=soma_invalidation_const, 
-      anisotropy=anisotropy
-    )
+                              labels, DBF, np.array([root], dtype=np.uint32),
+                              scale=soma_invalidation_scale,
+                              const=soma_invalidation_const, 
+                              anisotropy=anisotropy)
+      invalid_vertices[root]=True
 
   paths = []
   valid_labels = np.count_nonzero(labels)
     
- 
-
-  invalid_vertices = {}
-
   while valid_labels > 0:
     target = igneous.skeletontricks.find_target(labels, PDRF)
     path = igneous.dijkstra.path_from_parents(parents, target)
+    if soma_mode:
+      d = np.linalg.norm(anisotropy*(path - root),axis=1)
```

Comment:
Maybe a more descriptive variable name than `d`. I was confused for a moment thinking it was a single vertex. `path_to_root_dist` maybe?

Reply:
fixed!